test/integration/diag_nodes_test.go

```diff
@@ -44,7 +47,9 @@ func TestDiagNodeConditions(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 	nodeDiag := clusterdiags.NodeDefinitions{KubeClient: client}
```

Comment:
if you know the name of the all-in-one node (which we probably should get back with the node config from the start all in one helper), the first wait check can become:

```
	err = wait.Poll(200*time.Millisecond, 5*time.Second, func() (bool, error) {
		if _, err := client.Nodes().Get(nodeName); kapierror.IsNotFound(err) {
			return false, nil
		} else {
                  return true, err
                }
	})
	if err != nil {
		t.Errorf("unexpected error: %v", err)
	}
```

and the second:
```
	err = wait.Poll(200*time.Millisecond, time.Second, func() (bool, error) {
                if node, err := client.Nodes().Get(nodeName)
                if err != nil {
                  return false, err
                }
                node.Unschedulable = true
		if _, err := client.Nodes().Update(node); kapierror.IsConflict(err) {
			return false, nil
		} else {
                       return true, err
                } 
	})
	if err != nil {
		t.Errorf("unexpected error: %v", err)
	}
```